tests/Rules/PHPUnit/data/data-provider-declaration.php

```diff
@@ -0,0 +1,70 @@
+<?php declare(strict_types=1);
+
+namespace ExampleTestCase;
+
+class FooTestCase extends \PHPUnit\Framework\TestCase
+{
+	/**
+	 * @dataProvider provideBar Comment.
+	 * @dataProvider providebaz
+	 * @dataProvider provideQux
+	 * @dataProvider provideQuux
+	 */
+	public function testIsNotFoo(string $subject): void
```

Comment:
it would be great, if the rule could validate, whether the dataProviders return a correctly typed iterable in comparison to the expected parameter types of the test.

Reply:
implemented it in https://github.com/phpstan/phpstan-phpunit/pull/152